t/run/runenv_hashseed.t

```diff
@@ -16,121 +31,193 @@ skip_all_without_config('d_fork');
-      (is_miniperl() ? '' : 'num_buckets(%h),hash_traversal_mask(%h),') .
-      ' join "", @k;'
+    'my %h; keys(%h)=' . $min_buckets . '; ' .
+    '@h{' . $key_expr . '}=(); @k=keys %h; ' .
+      'print join ":", 0+@k, ' .
+      (is_miniperl() ? '' :  # no Hash::Util here!
+          'num_buckets(%h),hash_traversal_mask(\\%h), ') .
+      'join ",", @k;'
 ];
+
 for my $mode (@modes) {
     my $base_opts = {
         PERL_PERTURB_KEYS => $mode,
         PERL_HASH_SEED_DEBUG => 1,  # needed for non DEBUGGING builds
     };
+    my $descr_mode = sprintf "PERL_PERTURB_KEYS = %s", $mode;
+    my $random_same = 0;
+    my $seed_change_same = 0;
     for my $try (1 .. $repeat) {
-        my $descr = sprintf "PERL_PERTURB_KEYS = %s, try %2d:", $mode, $try;
-        my ( $out, $err )
+        my $descr = sprintf "%s, try %2d:", $descr_mode, $try;
+        my ( $out, $err, $seed )
             = my_runperl_and_capture( $base_opts, $print_keys );
-        $err =~ /HASH_SEED = (0x[a-f0-9]+)/
-            or die "Failed to extract hash seed from runperl_and_capture";
-        my $seed = $1;
         my $run_opts = { %$base_opts, PERL_HASH_SEED => $seed };
 
+        {
+            # first test that if we munge the seed we get a
+            # different result
+            my $munged_seed = $seed;
+            substr($munged_seed,-1)=~tr/0-9a-f/1-9a-f0/;
+            if ($munged_seed eq $seed) {
+                die "Failed to munge seed '$seed'";
+            }
+
+            local $run_opts->{PERL_HASH_SEED}= $munged_seed;
+            my ( $new_out, $new_err, $new_seed )
+                = my_runperl_and_capture( $run_opts, $print_keys );
+            if ($new_seed ne $munged_seed) {
+                die "Seed change didn't seem to propagate";
+            }
+            if ($out eq $new_out) {
+                # there is at least a 1/131072 chance that we got the
+                # same result.
+                $seed_change_same++;
+            }
+        }
+
         # now we have to run it again.
-        my ( $out1, $err1, $err_got_data1, $rand_bits_data1 )
+        my ( $out1, $err1, $seed1, $err_got_data1, $rand_bits1 )
             = my_runperl_and_capture( $run_opts, $print_keys );
 
         # and once more, these two should be the same
-        my ( $out2, $err2, $err_got_data2, $rand_bits_data2 )
+        my ( $out2, $err2, $seed2, $err_got_data2, $rand_bits2 )
             = my_runperl_and_capture( $run_opts, $print_keys );
 
-        if ( $mode eq 'RANDOM' ) {
-            isnt( $out1, $out2,
-                "$descr results in different key order with the same keys"
-            );
+        is( $err, $err2,
+            "$descr debug output was consistent between runs"
+        );
+        if( $seed1 ne $seed) {
+            die "Seed wasn't consistent on retry 1";
         }
-        else {
-            is( $out1, $out2,
-                "$descr results in the same key order each time"
-            );
+        if( $seed2 ne $seed) {
```

Comment:
Given the `$err2` output is the same (whether you end up comparing to `$err` or `$err1`), this test is redundant. Unless you also intended to remove the hash seed when parsing it out, in which case line 51 needs to change.